network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/AbstractBranchImpl.java

```diff
@@ -456,23 +460,26 @@ public <E extends Extension<T>> void addExtension(Class<? super E> type, E exten
     }
 
     @Override
-    public ConnectablePositionImpl<T> createConnectablePositionExtension(Feeder feeder, Feeder feeder1, Feeder feeder2,
-                                                                         Feeder feeder3) {
-        Objects.requireNonNull(feeder2);
-        if (feeder3 != null) {
-            throw new IllegalArgumentException("feeder3 must be null for branches");
-        }
+    public ConnectablePositionImpl<T> createConnectablePositionExtension(Feeder feeder, Feeder feeder1, Feeder feeder2, Feeder feeder3) {
         ConnectablePosition.check(feeder, feeder1, feeder2, feeder3);
-        ConnectablePositionAttributes cpa1 = ConnectablePositionAttributes.builder()
-                .label(feeder1.getName())
-                .order(feeder1.getOrder().orElse(null))
-                .direction(ConnectableDirection.valueOf(feeder1.getDirection().name()))
-                .build();
-        ConnectablePositionAttributes cpa2 = ConnectablePositionAttributes.builder()
-                .label(feeder2.getName())
-                .order(feeder2.getOrder().orElse(null))
-                .direction(ConnectableDirection.valueOf(feeder2.getDirection().name()))
-                .build();
+        ConnectablePositionAttributes cpa1 = null;
+        ConnectablePositionAttributes cpa2 = null;
+        if (feeder1 != null) {
+            cpa1 = ConnectablePositionAttributes.builder()
+                    .label(feeder1.getName())
+                    .order(feeder1.getOrder().orElse(null))
+                    .direction(ConnectableDirection.valueOf(feeder1.getDirection().name()))
+                    .build();
+        }
+
+        if (feeder2 != null) {
+            cpa2 = ConnectablePositionAttributes.builder()
+                    .label(feeder2.getName())
+                    .order(feeder2.getOrder().orElse(null))
+                    .direction(ConnectableDirection.valueOf(feeder2.getDirection().name()))
+                    .build();
+        }
+
         return new ConnectablePositionImpl<>(getBranch(),
                 null,
                 new ConnectablePositionImpl.FeederImpl(cpa1),
```

Comment:
null if cpa1 is null
null if cpa2 is null

and same thing in the AbstractBranchImpl constructor when initializing connectablePositionExtension

Reply:
in AbstractBranchImpl constructor there is already a check on cpa1 and cpa2